### DISCUSSION TITLE
[2주차] HTTP 지연은 왜 발생하는가?

### DISCUSSION THREAD

**Discussion by Hyevvy** [prgrms-web-devcourse/We-are-the-MASTERS-OF-HTTP#24] [Interview Question]
HTTP 트랜잭션 지연을 발생시키는 원인들을 간략하게 설명해주세요.

**Comment by punchdrunkard** [comment]
HTTP 지연을 발생시키는 원인은 2가지가 있습니다.
1. TCP 지연
2. HTTP 통신에서 커넥션이 순차적으로 처리될 경우 일어날 수 있는 지연.

1.의 지연은 TCP Slow Start 에서 주로 발생되며,
2.의 지연은 HTTP 트랜잭션이 일어날 때, 순차적인 통신을 하게 된다면 HTTP 트랜잭션을 위한 연걸을 수립하고 종료하는 과정에서 지연이 발생하게 됩니다.

이는 병렬  커넥션과 지속 커넥션을 통하여 해소할 수 있습니다.
또한, 지속 커넥션를 지원하기 위하여 HTTP 1.1 이전에는 keep-alive 헤더를 주로 이용하였으나
HTTP 1.1 부터는 기본적으로 지속 커넥션을 지원합니다.
